### PR TITLE
Neurodamus unified models

### PR DESCRIPTION
**Pull request by ferdonline**
## Context
Unified Neurodamus is the concept where all BBP simulation models are deployed together and can be loaded on demand by neurodamus-py.

This setup is desirable, not only because it doesn't require users to switch modules anymore, but also for developers and the ultimate goal of reproducibility, as it makes the whole set of models and their inter-dependencies stable and easier to deploy,

## This PR
Building on the new unified `neurodamus-models` repo in GitLab, we build a new Spack recipe which, extending the solid `neurodamus-model` recipe, builds the multiple models, side by side, to a common destination.

At the same time the `common` set of mods becomes a first-class model, since it it might be used by bglib-py, replacing the old `neurodamus-core+common_mods`.

### Impl details
 - Each model build output is stored in `<install_prefix>/lib/<model_name>`
 - Soft links to each binary `special` are created in bin, e.g. `bin/special-neocortex`, `bin/special-hippocampus`...
 - A new env var `MODELS_LIBRARY_PATH` points to `<install_prefix>/lib` to allow `neurodamus-py` to pick the requested one.
 - One prepends each model hocs dir to the env var `HOC_LIBRARY_PATH`. **We effectively make use of two hoc paths**,
 - python files which are required from within Hoc files have their path prepended to `PYTHONPATH` from the `common` model

### CODE REVIEW

File: var/spack/repos/builtin/packages/neurodamus3/package.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+from spack import *
+from spack.pkg.builtin.neurodamus_model import NeurodamusModel
+
+
+class Neurodamus3(NeurodamusModel):
```
Comment (jorblancoa) [inline var/spack/repos/builtin/packages/neurodamus3/package.py]: As i mentioned before, I think the name is a bit confusing. I would leave it as Neurodamus

**Review by jorblancoa — DISMISSED** [review]
LGTM!

File: var/spack/repos/builtin/packages/neurodamus3/package.py (hunk 1 above)
Comment (pramodk) [reply var/spack/repos/builtin/packages/neurodamus3/package.py]: Agree with Jorge - what `3` indicates here?

**Comment by ferdonline** [comment]
This PR is likely to be superseded because there is an increasing desire for multi-model builds.
Building on the same infrastructure, we might deploy a single "full" neurodamus with all mechanisms incorporated.
That is a significantly simpler design, more stable. The only drawback is that some effort needs to be done to **create models which 100% do not overlap**, which is being investigated.

**Pull request closed without merging**